src/core/execute.c

```diff
@@ -4909,6 +4909,19 @@ static int exec_child(
                         return r;
         }
 
+	if (context->memory_ksm) {
+#ifdef PR_SET_MEMORY_MERGE
+		r = prctl(PR_SET_MEMORY_MERGE, 1);
+		if (r < 0) {
+			*exit_status = EXIT_KSM;
+			return log_unit_error_errno(unit, r, "Failed to enable KSM %m");
+		}
```

Comment:
i am pretty sure this should not result in complete failure if not supported by the kernel. i.e. if ERRNO_IS_NOT_SUPPORTED(errno) holds, we should log and proceed. See how the NUMA  policy code handles this for example.

i.e. this is a very new feature, and the system should happily work without it, hence we should handle this gracefully

Reply:
I'll follow the pattern used in the NUMA policy code.